File: Backend/API/Controllers/Models/CreateJobInput.cs
namespace Backend.API.Controllers.Models
{
    public class CreateJobInput
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public int Deadline { get; set; }
        public double Payment { get; set; }
        public bool IsPaymentByHour { get; set; }
        public List<string> Skills { get; set; }
        public string ClientId { get; set; }

        public string? AssignedFreelancerId;

        public CreateJobInput(string title, string description, int deadline, double payment, bool isPaymentByHour, List<string> skills, string clientId, string assignedFreelancerId)
        {
            Title = title;
            Description = description;
            Deadline = deadline;
            Payment = payment;
            IsPaymentByHour = isPaymentByHour;
            Skills = skills;
            ClientId = clientId;
            AssignedFreelancerId = assignedFreelancerId;
        }
    }
}

File: frontend/src/models/CreateJobInput.ts
export default class CreateJobInput {
    Title: string;
    Description: string;
    Deadline: number;
    Payment: number;
    IsPaymentByHour: boolean;
    Skills: string[];
    ClientId: string;

    constructor(title: string, description: string, deadline: number, payment: number, isPaymentByHour: boolean, skills: string[], clientId: string) {
        this.Title = title;
        this.Description = description;
        this.Deadline = deadline;
        this.Payment = payment;
        this.IsPaymentByHour = isPaymentByHour;
        this.Skills = skills;
        this.ClientId = clientId;
    }
};